### PR TITLE
Basic pokemon detail build script

### PR DESCRIPTION
**Pull request by fabe85**
@vivek-sethia

### CODE REVIEW

File: scripts/build.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+];
+
+// when the choices of collection to be filled doesn't match, then exit the process
+if (possibleListeners.indexOf(collection) < 0) {
+    process.exit();
+}
+
+// listener for pokemons
+const listener = require(__base + 'app/controllers/filler/' + collection);
+listener.fill(function () {
```
Comment (vivek-sethia) [inline scripts/build.js]: change the names since it's not listener, change the name accordingly for build script.

File: app/controllers/filler/basicPokemonDetail.js [hunk 2]
```diff
@@ -4,11 +4,46 @@ const PokemonModel = require(__appbase + 'models/pokemon'),
...
+                var pokemonID, pokemonName, gender, male_ratio, female_ratio, breedable;
+                if(key != 'name' && typeof value.name !== "undefined") {
+                    pokemonID = key;
+                    pokemonName = value.name;
+                    gender = value.gender.abbreviation;
+                    male_ratio = value.gender.male_ratio;
+                    female_ratio = value.gender.female_ratio;
+                    breedable = value.gender.breedable;
 
+                    basicPokemonDetail.create({pokemonID: pokemonID, name: pokemonName,
```
Comment (vivek-sethia) [inline app/controllers/filler/basicPokemonDetail.js]: As discussed yesterday, database operations should be under separate file under `store` directory.

File: app/controllers/filler/basicPokemonDetail.js [hunk 3]
```diff
@@ -1,87 +1,31 @@
-const PokemonModel = require(__appbase + 'models/pokemon'),
-    PokemonStore = require(__appbase + 'stores/pokemon'),
-    jsonfile = require('jsonfile'),
-    async = require('async'),
-    cfg = require(__appbase + '../config');
-
 let fs = require('fs'),
```
Comment (vivek-sethia) [inline app/controllers/filler/basicPokemonDetail.js:3]: You have used `let, const` without using `"use strict";` and when I tried running the script it throws me an error.

File: app/stores/basicPokemonDetail.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            if (key != 'name' && typeof value.name !== "undefined") {
+                pokemonID = key;
+                pokemonName = value.name;
+                gender = value.gender.abbreviation;
+                male_ratio = value.gender.male_ratio;
+                female_ratio = value.gender.female_ratio;
+                breedable = value.gender.breedable;
+
+
+                basicPokemonDetail.create({
```
Comment (vivek-sethia) [inline app/stores/basicPokemonDetail.js]: Only db operations were needed to be moved to the stored other part could be still there in the filler/basicPokemondetails.js.  Its a suggestion since we usually keep database operations from the main file in thie being the filler//basicPokemondetails.js

**Comment by samitsv** [comment]
aren't we going to use pokemonGodata.json instead of the pokemonlist.json?

**Comment by sacdallago** [comment]
Guys this PR is old and conflicting.. ?

**Comment by swathi-ssunder** [comment]
@fabe85 was going to update this pull request. He should probably be done with this today.

**Comment by fabe85** [comment]
I adapted anything now to pokemonGoData.json. Please have a look.

File: app/models/BasicPokemonDetail.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
-let mongoose = require('mongoose');
+var mongoose = require('mongoose');
 
-let BasicPokemonDetailSchema = new mongoose.Schema({
+var BasicPokemonDetailSchema = new mongoose.Schema({
     pokemonID: {
         type: Number,
         unique: true,
```
Comment (swathi-ssunder) [inline app/models/BasicPokemonDetail.js:8]: Is this change necessary?

File: app/models/pokemonGoBasic.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    MaxHP: Number,
+    gender: {
+        abbreviation: {
+            type: String, enum: ['m', 'f', 'g', 'h'] //m=male, f=female, g=genderless, h=hermaphrodite
+        },
+        male_ratio: Number,
+        female_ratio: Number,
+        breedable: Boolean
+    }
+
```
Comment (swathi-ssunder) [inline app/models/pokemonGoBasic.js]: Can we follow one format for casing(the schema attributes)? Here, there are camelCase, snake_case, PascalCase... 😕

File: app/models/BasicPokemonDetail.js (hunk 5 above)
Comment (fabe85) [reply app/models/BasicPokemonDetail.js:8]: No you are right, I forgot to delete it completely because now it is obsolete.

File: app/models/pokemonGoBasic.js (hunk 6 above)
Comment (fabe85) [reply app/models/pokemonGoBasic.js]: Which format do you prefer? Because I only want to rebuild that once as the whose pokemonGoData.json has to be adapted.

File: app/models/BasicPokemonDetail.js (hunk 5 above)
Comment (fabe85) [reply app/models/BasicPokemonDetail.js:8]: I just realized that there is even a bunch of outdated files to be deleted. Maybe it is better to create a separate branch for cleaning things.

File: app/models/pokemonGoBasic.js (hunk 6 above)
Comment (vivek-sethia) [reply app/models/pokemonGoBasic.js]: For schema you can use camel-casing as we have followed the same in pokemonSighting. For e.g., Fast_Attacks to fastAttacks

File: app/models/pokemon.js [hunk 7]
```diff
@@ -1,28 +1,39 @@
...
-    weight     : {type: Number},
-    height     : {type: Number},
-    move       : {type: [Boolean]},
-    evolution  : {type: [String]},
-    pre_form   : {type: [String]}
 });
 
-module.exports = mongoose.model('pokemon', pokemonSchema);
\ No newline at end of file
+module.exports = mongoose.model('BasicPokemonDetail', pokemonGoBasicSchema);
```
Comment (vivek-sethia) [inline app/models/pokemon.js]: You can also name this to pokemon.

File: app/models/pokemon.js [hunk 8]
```diff
@@ -1,28 +1,39 @@
...
+var Schema = mongoose.Schema;
 
+var pokemonGoBasicSchema = new Schema({
+    Number: {type: String},
+    Name: {type: String},
+    Classification: {type: String},
+    Types: [String],
+    Resistant: [String],
+    Weaknesses: [String],
+    Fast_Attacks: [{Name: {type: String}, Type: {type: String}, Damage: {type: Number}}],
```
Comment (vivek-sethia) [inline app/models/pokemon.js]: For schema you can use camel-casing as we have followed the same in pokemonSighting. For e.g., Fast_Attacks to fastAttacks & Weaknesses to weaknesses

File: scripts/build.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+require(__dirname + '/../' + 'constants');
+
+if (collection === undefined) {
+    process.exit();
+}
+logger.info(collection);
+/*
+ * choices of collection to be filled
+ */
+var possibleListeners = [
```
Comment (vivek-sethia) [inline scripts/build.js]: variable name could be possibleCollections, since it's not listener.

File: app/models/pokemon.js (hunk 7 above)
Comment (fabe85) [reply app/models/pokemon.js]: Oh no wait because then the collection would not be called basicpokemondetails as required for the api

File: scripts/build.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    process.exit();
+}
+
+database.connect(function (db) {
+    db.on('open', function () {
+        // builder basic pokemon details
+        const builder = require(__base + 'app/controllers/filler/' + collection);
+        builder.insertToDb(function () {
+            process.exit();
+        });
```
Comment (swathi-ssunder) [inline scripts/build.js:27]: @fabe85 , @vivek-sethia - Is process.exit okay here? I think it was discussed to have a return instead.

Comment (fabe85) [reply scripts/build.js:27]: Yes you are right, sorry I did forget that
